chia/data_layer/data_store.py

```diff
@@ -751,10 +751,14 @@ async def get_internal_nodes(self, store_id: bytes32, root_hash: Optional[bytes3
         return internal_nodes
 
     async def get_keys_values_cursor(
-        self, reader: aiosqlite.Connection, root_hash: Optional[bytes32]
+        self,
+        reader: aiosqlite.Connection,
+        root_hash: Optional[bytes32],
+        only_keys: bool = False,
     ) -> aiosqlite.Cursor:
+        select_clause = "SELECT hash, key" if only_keys else "SELECT *"
         return await reader.execute(
-            """
+            f"""
             WITH RECURSIVE
                 tree_from_root_hash(hash, node_type, left, right, key, value, depth, rights) AS (
```

Comment:
Perhaps some code duplication would be worth it here, to do the recursive query without including value field? Not sure what the impact would be, given that I don't select value in the final query. (so we won't deal with it in the python side).

Reply:
This is already being made an f-string so could just use that.  I think?  `f""".... key, {maybe_value}depth,..."""`